### ISSUE TITLE
Update Amazon Payments to support foreign currencies

### ISSUE THREAD

**Issue by dgoldak** [easydigitaldownloads/easy-digital-downloads#6976]
Originally noted in https://github.com/easydigitaldownloads/easy-digital-downloads/issues/6832 but wanted to create a separate request for this.

You can't accept USD payments with an EU Amazon account. Customer talked to their support reps and this is what their last message was:

It is simply not possible to accept USD with an European Amazon Pay account and you cannot set up multiple currencies. Easy Digital Downloads needs to have a 'foreign currency feature' compatible with Amazon Pay,

That's the link to their developer Guide and the integration of Amazon Pay:

https://pay.amazon.com/de/developer/documentation/lpwa/201909330.

Related ticket: https://secure.helpscout.net/conversation/628055549/86677?folderId=2048992

cc @misulicus

**Comment by cklosowski** [comment]
@dgoldak @misulicus when you say "Foreign Currency Feature" does that mean that EDD supports multiple currencies on the same site?

Are all of the people talking about this issue using the currency switcher plugin? Out of the box, EDD supports a single currency, and we don't have any multi-currency plugins built inhouse. If the issue is that someone is using the currency converter plugin to accept currencies in a different value than that of their Amazon store, I'm not sure how _we_ are supposed to know what to accept it as.

**Comment by pippinsplugins** [comment]
I think I've figured out how to do this and it's ready for testing. I've also sent this off to a customer for testing.

**Comment by davidsherlock** [comment]
The customer has reported that this fix has not been resolved as per the ticket here:

https://secure.helpscout.net/conversation/739265998/97278?folderId=1847245

**Comment by davidsherlock** [comment]
The payment is being created then marked as "failed", however, it appears commissions are being recorded which _should_ only happen when the payment is successful.

**Comment by davidsherlock** [comment]
My mistake, the payments are marked as `completed` even though the funds are not within the store owners Amazon merchant account. The store owner then sets the payment to `failed` manually.

**Comment by pippinsplugins** [comment]
I don't think this is an EDD core issue. I think this is actually due to the currency switch plugin not filtering the currency code.

We set the currency code by calling `edd_get_currency()`. The only reason the currency would be wrong there is if the multi currency plugin is not running when `edd_get_currency()` is called during the Amazon Payments setup.

**Comment by davidsherlock** [comment]
@pippinsplugins based on the ticket referenced above, the customer is not using a currency switching plugin. My understanding of the issue is as follows:

1. The store is set to a specific currency (in this case USD). 
2. A customer purchases a product and pays using a EU based Amazon account.
3. Amazon sends confirmation back to the site but the payment is now in euros, instead of dollars.
4. The transaction fails, because the currencies do not properly match.

**Comment by cklosowski** [comment]
I'm actually going to remove the milestone here. I've recently been contacted by the Amazon Pay team and I am going to work with them to do an assessment of our integration.
